File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Java/src/main/java/simulacionBandaMusica/factories/PercusionistaFactory.java
package simulacionBandaMusica.factories;

import simulacionBandaMusica.enums.TipoPercusion;
import simulacionBandaMusica.models.Percusionista;

import static simulacionBandaMusica.utils.sorteoDatos.*;

public class PercusionistaFactory {

    public static Percusionista create() {
        String name = randomName();
        int experienceYear = randomAnno();
        TipoPercusion tipoPercusion = randomTipoPercusion();
        return new Percusionista(name, "Percusión", experienceYear, tipoPercusion);
    }

}

File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Kotlin/src/main/kotlin/simulacionBandaMusica/factories/PercusionistaFactory.kt
package simulacionBandaMusica.factories

import simulacionBandaMusica.models.Musico
import simulacionBandaMusica.utils.sorteoDatos

object PercusionistaFactory {
    fun create(): Musico.Percusionista {
        val name = sorteoDatos.randomName()
        val experienceYear = sorteoDatos.randomAnno()
        val tipoPercusion = sorteoDatos.randomTipoPercusion()
        return Musico.Percusionista(name, "Percusión", experienceYear, tipoPercusion)
    }
}